test.js

```diff
@@ -146,6 +146,14 @@ test('Function.prototype.apply with the arguments object', t => {
   t.true(foo('[string|number] <object>', 'bar', {}));
 });
 
+test('allows `this` to be the typeConfig string', t => {
+  function foo () {
+    return argsert.apply('<null|object> [number]', arguments);
```

Comment:
Should the first argument to apply be `this`? 

`argsert.apply(argsert, '<null|object> [number]', arguments)?